goto_handler.go

```diff
@@ -0,0 +1,116 @@
+package migrate
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+func (m *Migrate) HandleDirtyState(path string) {
+	// Perform actions when dirty is true
+		lastSuccessfulMigrationPath := filepath.Join(path, "lastSuccessfulmigration")
+		lastVersionBytes, err := os.ReadFile(lastSuccessfulMigrationPath)
+		if err != nil {
+			log.Fatal(fmt.Errorf("failed to read last successful migration file: %s", err))
+		}
+		lastVersionStr := strings.TrimSpace(string(lastVersionBytes))
+		lastVersion, err := strconv.ParseUint(lastVersionStr, 10, 64)
+		if err != nil {
+			log.Fatal(fmt.Errorf("failed to parse last successful migration version: %s", err))
+		}
+
+		if err := m.Force(int(lastVersion)) ; err != nil {
+			log.Fatal(fmt.Errorf("failed to apply last successful migration: %s", err))
+		}
+
+		fmt.Println("Last successful migration applied")
```

Comment:
Do you plan to keep those logs? If yes I'd suggest to use `log.verbose` and `log.Println` as done in the vanilla code.

Reply:
Note that the existing pattern is to check the verbosity first, one example:
```go
if log.verbose {
	log.Println("Finished after", time.Since(startTime))
}
```